libs/wingsdk/src/target-sim/bucket.inflight.ts

```diff
@@ -208,6 +208,32 @@ export class Bucket implements IBucketClient, ISimulatorResourceInstance {
     });
   }
 
+public async signed_url(key: string, duration?: Duration){
+  return this.context.withTrace({
+    message: `Signed URL (key=${key})`,
+    activity: async () =>{
+    if(!this.objectKeys.has(key)){
+        throw new Error(
+          `Cannot provide signed url for an non-existent key (key=${key})`
+        ); 
+    }
+    /**
+     * Generating a mock signed url response
+     */
+    const expiresInSeconds= duration?.seconds||(24*3600);
+    const params= new URLSearchParams();
+    params.set("Bucket","my_bucket");
+    params.set("key",key);
+    params.set("Expires",String(expiresInSeconds));
```

Comment:
how do we know when something is truly expiring? potentially we can have a URL of 2 minutes, and retrieve it 3 minutes later when nothing stops us. 
When designing the expiration, please take in mind that we initiate the class within each compilation...